homeassistant/scripts/benchmark/__init__.py

```diff
@@ -178,16 +179,90 @@ async def _logbook_filtering(hass, last_changed, last_updated):
 
     entity_attr_cache = logbook.EntityAttributeCache(hass)
 
-    def yield_events(event):
-        # pylint: disable=protected-access
-        entities_filter = logbook._generate_filter_from_config({})
+    entities_filter = convert_include_exclude_filter(
+        logbook.INCLUDE_EXCLUDE_BASE_FILTER_SCHEMA({})
+    )
+
+    def yield_events(event, entities_filter):
         for _ in range(10 ** 5):
+            # pylint: disable=protected-access
             if logbook._keep_event(hass, event, entities_filter, entity_attr_cache):
                 yield event
 
     start = timer()
 
-    list(logbook.humanify(hass, yield_events(event), entity_attr_cache))
+    list(
+        logbook.humanify(hass, yield_events(event, entities_filter), entity_attr_cache)
+    )
+
+    return timer() - start
+
+
+@benchmark
+async def filtering_entity_id(hass):
+    """Run a 100k state changes through entity filter."""
+    config = {
+        "include": {
+            "domains": [
+                "automation",
+                "script",
+                "group",
+                "media_player",
+                "custom_component",
+            ],
+            "entity_globs": [
+                "binary_sensor.*_contact",
+                "binary_sensor.*_occupancy",
+                "binary_sensor.*_detected",
+                "binary_sensor.*_active",
+                "input_*",
+                "device_tracker.*_phone",
+                "switch.*_light",
+                "binary_sensor.*_charging",
+                "binary_sensor.*_lock",
+                "binary_sensor.*_connected",
+            ],
+            "entities": [
+                "test.entity_1",
+                "test.entity_2",
+                "binary_sensor.garage_door_open",
+                "test.entity_3",
+                "test.entity_4",
+            ],
+        },
+        "exclude": {
+            "domains": ["input_number"],
+            "entity_globs": ["media_player.google_*", "group.all_*"],
+            "entities": [],
+        },
+    }
+
+    entity_ids = [
+        "automation.home_arrival",
+        "script.shut_off_house",
+        "binary_sensor.garage_door_open",
+        "binary_sensor.front_door_lock",
+        "binary_sensor.kitchen_motion_sensor_occupancy",
+        "switch.desk_lamp",
+        "light.dining_room",
+        "input_boolean.guest_staying_over",
+        "person.eleanor_fant",
+        "alert.issue_at_home",
+        "calendar.eleanor_fant_s_calendar",
+        "sun.sun",
+    ]
+
+    def yield_entity_ids(entity_ids, config):
+        entities_filter = convert_include_exclude_filter(config)
+        size = len(entity_ids)
+        for i in range(10 ** 5):
+            entity_id = entity_ids[i % size]
+            if entities_filter(entity_id):
+                yield entity_id
+
+    start = timer()
+
+    list(yield_entity_ids(entity_ids, config))
```

Comment:
```suggestion
    entities_filter = convert_include_exclude_filter(config)

    start = timer()

    size = len(entity_ids)
    for i in range(10 ** 5):
        entities_filter(entity_ids[i % size]):
```

We should make this as simple as possible to focus the benchmark on `entities_filter`

Reply:
```
# hass --script benchmark filtering_entity_id
Using event loop: asyncio.unix_events
Benchmark filtering_entity_id done in 0.3853463870473206s
Benchmark filtering_entity_id done in 0.38362657208926976s
Benchmark filtering_entity_id done in 0.38990459381602705s
Benchmark filtering_entity_id done in 0.383747274056077s
```